src/services/room-service.js

```diff
@@ -18,4 +20,32 @@ export default class RoomService {
     await this.roomsStore.save(newRoom);
     return newRoom;
   }
+
+  async getRoomDetailsById(roomId) {
+    const roomDetails = await this.roomsStore.findOne({ _id: roomId });
+
+    if (!roomDetails) {
+      return null;
+    }
+
+    const owner = await this.userService.getUserById(roomDetails.owner);
+    roomDetails.owner = {
+      userId: owner._id,
+      username: owner.username,
+      email: owner.email
+    };
+
+    const members = await this.userService.getUsersByIds(roomDetails.members.map(member => member.userId));
+
+    roomDetails.members = roomDetails.members.map(member => {
+      const memberDetails = members.find(user => member.userId === user._id);
+      return {
+        ...member,
+        email: memberDetails.email,
+        username: memberDetails.username
+      };
+    });
+
```

Comment:
I'm a little bit unsure about this approach. Usually (and by "usually" I mean: in case of documents and revisions) the service returns more or less the raw data objects from the DB, and then the client data mapper resolves user IDs (`createdBy` etc.) to the actual user objects (with more or less user data being there based on the permissions of the current user). This approach now constructs the full data model for the UI inside the service. I'd like to know your opinion on whether it wouldn't make sense to do it here like we do with documents (given the fact that probably room members should not be able to see the email addresses of other room members).

Reply:
Moved to client-data-mapper.

Reply:
:+1: thanks